src/hooks/usePerfEventHandler.ts

```diff
@@ -0,0 +1,23 @@
+import { useAtomValue } from 'jotai/utils'
+import { useCallback } from 'react'
+import { swapEventHandlersAtom } from 'state/swap'
+import { PerfEventHandlers } from 'state/swap/perf'
+
+export function usePerfEventHandler<
+  K extends keyof PerfEventHandlers,
+  P extends Parameters<NonNullable<PerfEventHandlers[K]>>,
+  A extends P[0],
+  E extends Awaited<P[1]>,
+  H extends PerfEventHandlers[K] & ((args: A, event: Promise<E>) => void)
```

Comment:
can you add some comments to explain what these are

Reply:
I can name these with words rather than letters 🤷